### PR TITLE
Form config

### PR DESCRIPTION
**Pull request by ciremusyoka**
Use openSrp form config package to:
- Upload config form
- Edit config form
- Download uploaded config forms
- Publish manifest files

issue #900

### CODE REVIEW

File: src/configs/lang.ts [hunk 1]
```diff
@@ -573,6 +573,36 @@ export const ALB_TABLETS_DISTRIBUTED = translate(
...
+export const FORM_REQUIRED_LABEL = translate('FORM_REQUIRED_LABEL', 'Form is required');
+export const FORM_NAME_REQUIRED_LABEL = translate(
+  'FORM_NAME_REQUIRED_LABEL',
+  'Form name is required'
+);
+export const FIND_RELEASE_LABEL = translate('FIND_RELEASE_LABEL', 'Find Releases');
+export const FIND_DRAFT_FILE_LABEL = translate('FIND_DRAFT_FILE_LABEL', 'Find Draft Files');
+export const FIND_RELEASE_FILES = translate('FIND_RELEASE_FILES', 'Find Release Files');
+export const FIND_VALIDATOR_FILES = translate('FIND_VALIDATOR_FILES', 'Find Validator Files');
+
```
Comment (moshthepitt) [inline src/configs/lang.ts:605]: These need to be added to en.json

File: src/containers/pages/ConfigForm/manifest/filesList/index.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+ */
+const mapStateToProps = (
+  _: Partial<Store>,
+  ownProps: RouteComponentProps<RouteParams>
+): ManifestFilesProps => {
+  const formVersion = ownProps.match.params.id || '';
+  return {
+    formVersion,
+  };
+};
```
Comment (moshthepitt) [inline src/containers/pages/ConfigForm/manifest/filesList/index.tsx:103]: What's the use of getting this formVersion in mapStateToProps?  If its coming from props, just get it in the component?  Ama there is some other reason?

Comment (ciremusyoka) [reply src/containers/pages/ConfigForm/manifest/filesList/index.tsx:103]: @moshthepitt from my understanding `RouterParams` are passed only to the top level component  so it is either i pass the formVersion or i pass all router params. Just choose to pass the form version alone.

Comment (moshthepitt) [reply src/containers/pages/ConfigForm/manifest/filesList/index.tsx:103]: I think the component has access to these router props.  See https://github.com/onaio/reveal-frontend/blob/72abd6d96d181d2b2c4d8d722fdf49355096322d/src/containers/pages/IRS/Map/index.tsx#L126 and https://github.com/onaio/reveal-frontend/pull/929/files#diff-0a00081136a35b6f7d1a2672959bfbedR67 but you may need to extend the props like this https://github.com/onaio/reveal-frontend/pull/929/files#diff-8825a081fc6f0bc30cb3fd21223bbceeR29

File: src/configs/lang.ts (hunk 1 above)
Comment (ciremusyoka) [reply src/configs/lang.ts:605]: done

**Comment by p-netm** [comment]
@ciremusyoka can one pass tableProps to the containers through the exposed package interfaces, Or have you found a way to address the default no Data component look

**Comment by ciremusyoka** [comment]
> @ciremusyoka can one pass tableProps to the containers through the exposed package interfaces, Or have you found a way to address the default no Data component look

@p-netm currently it is not possible but will address that on issue #958

**Comment by Naima-Bashir** [comment]
Upload config form   --passed QA
Edit config form    --passed QA
Download uploaded config forms    --passed QA
Publish manifest files      --passed QA

**Pull request merged**